internal/hierarchy/hierarchy.go

```diff
@@ -110,3 +126,19 @@ func Group(addrs []resolver.Address) map[string][]resolver.Address {
 	}
 	return ret
 }
+
+// GroupEndpoints groups endpoints as per Group above.
```

Comment:
Pet peeve: please don't use "as per" in our docs. :)  Just "as" or "per" is probably what you want instead.

Except in this case, we want to just copy/paste the whole thing and update as appropriate, because we'll be deleting the other one.

Reply:
What's wrong with as per lol?

Rewrote comment to be like the one above except for discussing endpoints.

Reply:
> What's wrong with as per lol?

It's always redundant.  Just "as" or "per" is sufficient.  https://english.stackexchange.com/a/54880

But also personal preference, I guess.  Some people like saying "ATM machine" and "PIN number" too.

Reply:
Oh lol so it's redundant.